index.js

```diff
@@ -67,16 +67,20 @@ class SHACLValidator {
   }
 
   // Exposed to be available from validation functions as `SHACL.nodeConformsToShape`
-  nodeConformsToShape(focusNode, shapeNode) {
+  nodeConformsToShape(focusNode, shapeNode, engine = this.validationEngine.clone()) {
```

Comment:
Why clone?

Reply:
using a fresh instance of validation engine we avoid side effects (collection of result details) on the current instance.
Only when calling from a `sh:node` constraint we do want to affect the current instance, in the other cases (`sh:and`, `sh:or`, `sh:not`...) we are only interested in the boolean result and we don't care about the details.

Reply:
Hm, I wonder if that is entirely true. I was always thinking that to improve the error messages in complex scenarios would require a smarter analysis of the nested results. But maybe it's better to reorganise the shapes and write good, targeted messages...

Reply:
admittedly, this PR is a bit of hacking trying to fix the anomaly. Redesigning all the result details to provide more insights for violations of logical constraints is intriguing but beyond the scope of this PR.